core/blockly.js

```diff
@@ -406,14 +406,25 @@ Blockly.hideChaff = function(opt_allowToolbox) {
  * .contentLeft: Offset of the left-most content from the x=0 coordinate.
  * .absoluteTop: Top-edge of view.
  * .absoluteLeft: Left-edge of view.
+ * .toolboxWidth: Width of toolbox, if it exists.  Otherwise zero.
+ * .toolboxHeight: Height of toolbox, if it exists.  Otherwise zero.
+ * .flyoutWidth: Width of the flyout if it is always open.  Otherwise zero.
+ * .flyoutHeight: Height of flyout if it is always open.  Otherwise zero.
+ * .toolboxPosition: Top, bottom, left or right.
  * @return {Object} Contains size and position metrics of main workspace.
  * @private
  * @this Blockly.WorkspaceSvg
  */
 Blockly.getMainWorkspaceMetrics_ = function() {
   var svgSize = Blockly.svgSize(this.getParentSvg());
   if (this.toolbox_) {
-    svgSize.width -= this.toolbox_.width;
+    // If the toolbox is at the bottom it's laid out separately from the main
```

Comment:
Guess the comment's not clear enough, then.

In the case that the toolbox is at the top it seems to overlap the main workspace and therefore I need to subtract its height.  Also in the left and right cases.  But in the case where it's at the bottom it doesn't overlap, so I don't need to subtract its height.

...but if that's the case, then the else should be an else if, covering left and right, so that nothing is subtracted if it's at bottom.